script/deploy/healthCheck.ts

```diff
@@ -239,278 +239,356 @@ const main = defineCommand({
     }
 
     //          ╭─────────────────────────────────────────────────────────╮
-    //          │                   Check approved DEXs                   │
+    //          │                   Check whitelisted addresses           │
     //          ╰─────────────────────────────────────────────────────────╯
-    if (dexs) {
-      consola.box('Checking DEXs approved in diamond...')
+    if (expectedWhitelistedAddresses) {
+      consola.box('Checking whitelisted addresses in diamond...')
 
-      // connect with diamond to get whitelisted DEXs
-      const dexManager = getContract({
+      // connect with diamond to get whitelisted addresses
+      const whitelistManager = getContract({
         address: deployedContracts['LiFiDiamond'],
         abi: parseAbi([
-          'function approvedDexs() external view returns (address[])',
-          'function isFunctionApproved(bytes4) external returns (bool)',
+          'function getWhitelistedAddresses() external view returns (address[])',
+          'function isFunctionApproved(bytes4) external view returns (bool)',
+          'function getApprovedFunctionSignatures() external view returns (bytes4[])',
         ]),
         client: publicClient,
       })
 
-      const approvedDexs = await dexManager.read.approvedDexs()
-
-      let numMissing = 0
+      let onChainWhitelisted: Address[] = []
+      try {
+        onChainWhitelisted =
+          await whitelistManager.read.getWhitelistedAddresses()
+      } catch (error) {
+        logError('Failed to get whitelisted addresses from chain')
+        // Don't skip the checks - we'll still check against the config
+      }
 
-      // Check for each address in dexs.json if it is whitelisted
-      for (const dex of dexs) {
-        if (!dex) {
-          logError(`Encountered undefined DEX address.`)
-          continue
-        }
+      // Check if there are any expected whitelisted addresses in config
+      if (
+        expectedWhitelistedAddresses &&
+        expectedWhitelistedAddresses.length > 0
+      ) {
+        let numMissing = 0
+
+        // If we couldn't get on-chain addresses, we can only check the config addresses
```

Comment:
I am not sure if it makes sense to continue if we cannot get any whitelisted addresses from chain....could you pls provide arguments for why this makes sense? 

And is onChainWhitelisted always an array (can it not be null/undefined)?

Reply:
That's true then empty array will be returned. I’ve updated the script by removing the first part. script compares all whitelisted addresses from the config against the on-chain whitelisted addresses (even when it's empty). If an address from the config is missing on-chain it logs an error. Similarly if there’s an onchain whitelisted address that’s not present in the config,it also logs an error